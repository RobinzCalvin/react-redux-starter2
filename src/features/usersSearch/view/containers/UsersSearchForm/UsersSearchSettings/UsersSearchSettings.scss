@import '~shared/styles/mixins';

.users-search-settings {
  width: 11rem;

  @include breakpoint-up(sm) {
    width: 27rem;
  }

  &__row {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin-bottom: 1.5rem;

    @include breakpoint-up(sm) {
      flex-direction: row;
      margin-bottom: 2.5rem;
    }

    &:last-child {
      margin-bottom: 0;
    }
  }

  &__item {
    margin-bottom: 1rem;
    width: 11rem;

    @include breakpoint-up(sm) {
      margin-bottom: 0;
    }
  }

  &__repos-number {
    display: flex;
    justify-content: space-between;
    margin-top: 0.7rem;
    width: 11rem;

    &-input {
      width: 5rem;
    }
  }

  &__settings-group {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }
}
