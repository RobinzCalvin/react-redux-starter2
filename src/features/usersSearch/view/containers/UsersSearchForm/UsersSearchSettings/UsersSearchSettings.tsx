import * as React from 'react';
import block from 'bem-cn';

import { FormLabel } from 'shared/view/elements';
import { TextInputField, SelectField, NumberInputField, RadioField } from 'shared/view/form';

import { searchByOptions, perPageOptions, fieldNames } from '../constants';
import './UsersSearchSettings.scss';

const b = block('users-search-settings');

function UsersSearchSettings() {
  return (
    <div className={b()}>
      {renderFirstRowSettings()}
      {renderSecondRowSettings()}
    </div>
  );

  function renderFirstRowSettings() {
    return (
      <div className={b('row')}>
        <div className={b('item')}>
          <SelectField options={searchByOptions} label="Search by" name={fieldNames.searchBy} fullWidth />
        </div>
        <div className={b('item')}>
          <TextInputField name={fieldNames.reposLanguage} label="Repositories language" />
        </div>
      </div>
    );
  }

  function renderSecondRowSettings() {
    return (
      <div className={b('row')}>
        <div className={b('item')}>
          {renderSearchTypeSettings()}
        </div>
        <div className={b('settings-group')}>
          <div className={b('item')}>
            {renderRepositoriesNumberSettings()}
          </div>
          <div className={b('item')}>
            <SelectField options={perPageOptions} label="Results per page" name={fieldNames.perPage} fullWidth />
          </div>
        </div>
      </div>
    );
  }

  function renderSearchTypeSettings() {
    return (
      <FormLabel>
        Search type
        <div className={b('checkbox-group')}>
          <RadioField name={fieldNames.searchType} value="user" label="User" />
          <RadioField name={fieldNames.searchType} value="org" label="Organizations" />
          <RadioField name={fieldNames.searchType} value="both" label="Both" />
        </div>
      </FormLabel>
    );
  }

  function renderRepositoriesNumberSettings() {
    return (
      <FormLabel>
        Repositories number
        <div className={b('repos-number')}>
          <div className={b('repos-number-input')}>
            <NumberInputField
              name={fieldNames.minRepos}
              label="min"
              fullWidth
            />
          </div>
          <div className={b('repos-number-input')}>
            <NumberInputField
              name={fieldNames.maxRepos}
              label="max"
              fullWidth
            />
          </div>
        </div>
      </FormLabel>
    );
  }
}

export default UsersSearchSettings;
