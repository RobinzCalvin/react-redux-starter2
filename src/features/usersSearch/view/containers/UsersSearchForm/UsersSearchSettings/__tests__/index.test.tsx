import React from 'react';
import { shallow } from 'enzyme';

import UsersSearchSettings from '../UsersSearchSettings';

describe('UsersSearchSettings component', () => {
  const component = shallow(<UsersSearchSettings />);
  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
