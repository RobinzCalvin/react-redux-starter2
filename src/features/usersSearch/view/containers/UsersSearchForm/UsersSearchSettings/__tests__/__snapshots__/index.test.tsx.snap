// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersSearchSettings component should match snapshot 1`] = `
<div
  className="users-search-settings"
>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <FieldWithComponent(SelectField)
        label="Search by"
        name="searchBy"
        options={
          Array [
            Object {
              "label": "Username & email",
              "value": "username-email",
            },
            Object {
              "label": "Username",
              "value": "login",
            },
            Object {
              "label": "Email",
              "value": "email",
            },
            Object {
              "label": "Full name",
              "value": "fullname",
            },
          ]
        }
      />
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <FieldWithComponent(TextInputField)
        label="Repositories language"
        name="reposLanguage"
      />
    </div>
  </div>
  <div
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <WithStyles(WithFormControlContext(FormLabel))>
        Search type
        <div
          className={
            Object {
              "has": [Function],
              "is": [Function],
              "mix": [Function],
              "split": [Function],
              "state": [Function],
              "toString": [Function],
            }
          }
        >
          <FieldWithComponent(RadioField)
            label="User"
            name="searchType"
            value="user"
          />
          <FieldWithComponent(RadioField)
            label="Organizations"
            name="searchType"
            value="org"
          />
          <FieldWithComponent(RadioField)
            label="Both"
            name="searchType"
            value="both"
          />
        </div>
      </WithStyles(WithFormControlContext(FormLabel))>
    </div>
    <div
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
    >
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <WithStyles(WithFormControlContext(FormLabel))>
          Repositories number
          <div
            className={
              Object {
                "has": [Function],
                "is": [Function],
                "mix": [Function],
                "split": [Function],
                "state": [Function],
                "toString": [Function],
              }
            }
          >
            <div
              className={
                Object {
                  "has": [Function],
                  "is": [Function],
                  "mix": [Function],
                  "split": [Function],
                  "state": [Function],
                  "toString": [Function],
                }
              }
            >
              <FieldWithComponent(NumberInputField)
                label="min"
                name="minRepos"
              />
            </div>
            <div
              className={
                Object {
                  "has": [Function],
                  "is": [Function],
                  "mix": [Function],
                  "split": [Function],
                  "state": [Function],
                  "toString": [Function],
                }
              }
            >
              <FieldWithComponent(NumberInputField)
                label="max"
                name="maxRepos"
              />
            </div>
          </div>
        </WithStyles(WithFormControlContext(FormLabel))>
      </div>
      <div
        className={
          Object {
            "has": [Function],
            "is": [Function],
            "mix": [Function],
            "split": [Function],
            "state": [Function],
            "toString": [Function],
          }
        }
      >
        <FieldWithComponent(SelectField)
          label="Results per page"
          name="perPage"
          options={
            Array [
              Object {
                "label": 30,
                "value": 30,
              },
              Object {
                "label": 50,
                "value": 50,
              },
              Object {
                "label": 100,
                "value": 100,
              },
            ]
          }
        />
      </div>
    </div>
  </div>
</div>
`;
